packages/next-dashboard/src/components/charts/WrapChart.js

```diff
@@ -1,14 +1,28 @@
 // @flow
+/*:: import * as R from 'react'; */
 
 // Look, I'm technically hooking into some recharts magic.
 // They have a bug in that the cx/cy calculation is done differently
 // for radial charts and piecharts, so I'm hooking into their
 // magic here, with some weird ass typing, just to recalculate cy/cx
 // in the same way in all charts.
 
-// eslint-disable-next-line flowtype/no-weak-types
-const WrapChart = ({ children, height, width }: any) => {
-  return children(width, height);
+type Props = {
+  children: (number, number) => R.Node,
+  height?: number,
+  width?: number,
+  hoffset?: number,
+  voffset?: number,
+};
+
+const WrapChart = ({
+  children,
+  height = 0,
+  width = 0,
+  hoffset = 0,
+  voffset = 0,
+}: Props) => {
+  return children(width - voffset, height - hoffset);
 };
```

Comment:
I still don't really understand why this component is necessary and what bug it circumvents. I think it's kind of... awkward and would be nice to get rid of this component entirely in the future. Presumably that is the idea as it's meant to be a workaround of some issue, but it would be nice to know/understand what the issue is in the first place.

Pointing it out now since if we're adding more functionality to it, it'll be (a bit) trickier to replace in the future.

Reply:
It's so we can get the height/width calculated from RechartResponsiveContainer for internal compensations. A more sane solution would probably be to reimplement the responsive container ourselves with these options and stop using this hack, yes.

The added functionality is because something with how we were drawing the legend ended up outside the box and I needed somewhere to remove some height. It was previously done with an !important css rule which I felt was even more dirty.

Reply:
```css
.recharts-responsive-container {	
  height: calc(100% - 14px) !important;	
}
```
Specifically this is what the "hoffset" is meant to replace

Reply:
Alright, I guess so